File: app/src/main/java/eu/toma/dev/playground/warcraft/mounts/all/MVP.java
package eu.toma.dev.playground.warcraft.mounts.all;

import java.util.List;

import eu.toma.dev.playground.warcraft.mounts.Mount;

public interface MVP
{
    interface View
    {
        void onMounts(List<Mount> newMounts);

        void onMountsFetchError();

        void onNoMounts();

        void showLoading();

        void hideLoading();

        void showContent();

        void hideContent();
    }

    interface Presenter
    {
        void getMounts();
    }
}

File: app/src/main/kotlin/warcraft/mounts/all/MVP.kt
package warcraft.mounts.all

import warcraft.mounts.Mount

interface MVP {
    interface View {
        fun onMounts(newMounts: List<Mount>)

        fun onMountsFetchError()

        fun onNoMounts()

        fun showLoading()

        fun hideLoading()

        fun showContent()

        fun hideContent()
    }

    interface Presenter {
        fun getMounts()
    }
}
